test/LiquidDemocracy.test.js

```diff
@@ -10,53 +10,100 @@ const should = require('chai')
+        }
     });
 
-    it('should not allow unregistered voters to vote', async function () {
-      await assertRevert(this.liquidDemocracyBallot.vote(accounts[4], 0));
+    describe('registration set up', function () {
+        it('should register voters successfully', async function () {
+            const registered = await this.liquidDemocracyBallot.getVoterData(accounts[0]);
+            let registeredWeight = registered[0].toString();
+            let registeredStatus = registered[1];
+            const unregistered = await this.liquidDemocracyBallot.getVoterData(accounts[4]);
+            let unregisteredWeight = unregistered[0].toString();
+            let unregisteredStatus = unregistered[1];
+
+            registeredWeight.should.equal('1');
+            registeredStatus.should.equal(true);
+            unregisteredWeight.should.equal('0');
+            unregisteredStatus.should.equal(false);
+        });
     });
 
-    it('should not allow registered voters to vote more than once', async function () {
-      await this.liquidDemocracyBallot.vote(accounts[0], 0);
-      await assertRevert(this.liquidDemocracyBallot.vote(accounts[0], 0));
+    describe('voting', function () {
+        it('should allow registered voters to vote', async function () {
+            await this.liquidDemocracyBallot.vote(accounts[0], 0);
+            await this.liquidDemocracyBallot.vote(accounts[1], 0);
+            await this.liquidDemocracyBallot.vote(accounts[2], 1);
+            const voteCountOption0 = await this.liquidDemocracyBallot.getBallotVoteCount(0);
+            const voteCountOption1 = await this.liquidDemocracyBallot.getBallotVoteCount(1);
+
+            voteCountOption0.toNumber().should.equal(2);
+            voteCountOption1.toNumber().should.equal(1);
+        });
+
+        it('should not allow unregistered voters to vote', async function () {
+            await assertRevert(this.liquidDemocracyBallot.vote(accounts[4], 0));
+        });
+
+        it('should not allow registered voters to vote more than once', async function () {
+            await this.liquidDemocracyBallot.vote(accounts[0], 0);
+            await assertRevert(this.liquidDemocracyBallot.vote(accounts[0], 0));
+        });
     });
-  });
 
-  describe('delegating', function () {
-    // TODO
-  });
+    describe('delegations produce the expected output', function () {
+
+        it('should perform a simple delegation', async function () {
+            await this.liquidDemocracyBallot.delegate(accounts[1], { from: accounts[0] });
+
+            const representative = await this.liquidDemocracyBallot.getMyRepresentative({ from: accounts[0] })
+            const representativeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[1] })
+            const representeeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[0] })
+
+            assert.equal(representative, accounts[1]);
```

Comment:
We should probably be consistent on using `should` or `assert` throughout the test file. I was [reading about that](https://joshldavis.com/2013/05/27/difference-between-tdd-and-bdd/) before using `should` for the voting tests I added, it comes down to a somewhat blurry line between BDD vs TDD. I'm thinking we can stick with `should` as it's already in `master`, but would like to hear what you think.

Reply:
I think that's really not so important, they are just aliases, it's up to us to decide which ones to use. But I agree we should stick with the same one. I'm just used to use asserts in testing, but I'll change it to should to be consistent